### ISSUE TITLE
Database file created in /tmp despite db_location argument

### ISSUE THREAD

**Issue by fjansson** [UCL-CCS/EasyVVUQ#204]
The EasyVVUQ database file gets created in /tmp, and specifying alternative location does not work. /tmp is fine for testing but bad for production use, as files there may be automatically cleaned.

The /tmp location is set in campaign.py, line 217, unless db_location is specified.
```python
 elif self.db_type == 'json':
            from .db.json import CampaignDB
            if self.db_location is None:
                self.db_location = tempfile.mkstemp(suffix='.json', prefix='easyvvuq')[1]
```
The Campaign object takes a db_location argument, but it is overwritten here: https://github.com/UCL-CCS/EasyVVUQ/blob/e6e534770f8f797b6fc12c046f994b2126aee8c0/easyvvuq/campaign.py#L133

also db_type gets overwritten here: https://github.com/UCL-CCS/EasyVVUQ/blob/e6e534770f8f797b6fc12c046f994b2126aee8c0/easyvvuq/campaign.py#L134

Perhaps db_location could default to work_dir ?

**Comment by raar1** [comment]
Thanks for spotting this. Could you pull down the branch in PR #206 and check if the changes have fixed the problem?

**Comment by fjansson** [comment]
Testing the branch from PR #206. If I provide a *file name* to `db_location`, it works and the database is created in that file. I'd say the worst part of the bug is fixed.

Some remaining annoyances:
* the default is still /tmp
* one has to provide a file name as `db_location`, a directory is not enough.
* one has to make up a new unique name for the db_location, otherwise old databases get silently overwritten (I think)

I think it would be nice to just provide a work_dir, where each new run of EasyVVUQ creates its own
directory and database with unique random names. The run directories already work this way.

**Comment by raar1** [comment]
Thanks for testing. Yes, what you want it to do is actually what it is supposed to be doing. It appears there was a bug in the db_location setting that meant the campaign db was not being stored in the campaign dir. I think I've fixed that now (at least, when I tested it, the campaign dir was appearing in my local directory, and it contained the db).

Can you try with the new version of the PR branch please? Don't specify `db_location`, or `work_dir`. Just a simple `my_campaign = uq.Campaign(name='name-of-campaign')` _should_ now do what you want. Let me know if this has helped...

**Comment by fjansson** [comment]
It seems to be working nicely if I don't specify a db_type, with neither work_dir nor db_location, or with just work_dir.

With db_type='json', the database still ends up in /tmp, unless I provide db_location.
I think the code that needs changing is in campaign.py, around line 217.

**Comment by raar1** [comment]
Thanks again for testing this out. I have made further changes, so that the json db should now be getting stored in the same location as the sql one (i.e. in the campaigns dir where it should obviously go). In the process I found and fixed another bug to do with the json db not saving its state to file upon adding a sampler (although this is not relevant to your issue here). Hopefully we have exorcised mandatory tmp/ locations from the library now... ;)
